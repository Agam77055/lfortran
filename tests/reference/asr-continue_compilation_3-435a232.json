{
    "basename": "asr-continue_compilation_3-435a232",
    "cmd": "lfortran --semantics-only --continue-compilation --no-color {infile}",
    "infile": "tests/errors/continue_compilation_3.f90",
    "infile_hash": "47fd786aa24148260de63ab296703713419f3637f23da4a9010a2f6d",
    "outfile": null,
    "outfile_hash": null,
    "stdout": null,
    "stdout_hash": null,
    "stderr": "asr-continue_compilation_3-435a232.stderr",
    "stderr_hash": "0dbbcfb1f14f5ae594f1eb9cb9876a172a86732726a5227db4015d32",
    "returncode": 1
}