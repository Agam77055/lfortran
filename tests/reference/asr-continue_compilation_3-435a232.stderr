syntax error: Token '::' is unexpected here
  --> tests/errors/continue_compilation_3.f90:19:9
   |
19 |     rea :: test_real(12)
   |         ^^ 

syntax error: Token ')' is unexpected here
  --> tests/errors/continue_compilation_3.f90:20:21
   |
20 |     real :: test_re()
   |                     ^ 

syntax error: The LFortran pragma !LF$ must be followed by a space
   --> tests/errors/continue_compilation_3.f90:133:9
    |
133 |         !LF$unroll 4  ! Error: Missing space after `!LF$`
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 

semantic error: Module 'continue_compilation_3_fake_module' not declared in the current source and the modfile was not found
  --> tests/errors/continue_compilation_3.f90:10:5
   |
10 |     use continue_compilation_3_fake_module
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 

semantic error: Empty array constructor is not allowed
  --> tests/errors/continue_compilation_3.f90:53:9
   |
53 |     a = []
   |         ^^ 

semantic error: Empty array constructor is not allowed
  --> tests/errors/continue_compilation_3.f90:55:16
   |
55 |     print *, [[[], [[]]], [[]], []]
   |                ^^ 

semantic error: Empty array constructor is not allowed
  --> tests/errors/continue_compilation_3.f90:56:16
   |
56 |     print *, [[[], [[]]], []]
   |                ^^ 

semantic error: Rank mismatch in array reference: the array `b` has rank `1`, but is referenced as rank `2`
  --> tests/errors/continue_compilation_3.f90:58:5
   |
58 |     b(:,:) = 1
   |     ^^^^^^ 

semantic error: Rank mismatch in array reference: the array `b` has rank `1`, but is referenced as rank `2`
  --> tests/errors/continue_compilation_3.f90:59:5
   |
59 |     b(:,:) = 2
   |     ^^^^^^ 

semantic error: Type member xx is not an array so it cannot be indexed.
  --> tests/errors/continue_compilation_3.f90:61:5
   |
61 |     y%xx(:) = 1
   |     ^^^^^^^ 

semantic error: Rank mismatch in array reference: the array `str` has rank `1`, but is referenced as rank `2`
  --> tests/errors/continue_compilation_3.f90:63:5
   |
63 |     str(1, 2)(:) = '1234'
   |     ^^^^^^^^^^^^ 

semantic error: Rank mismatch in array reference: the array `str` has rank `1`, but is referenced as rank `3`
  --> tests/errors/continue_compilation_3.f90:64:5
   |
64 |     str(1,2,3)(:) = '1234'
   |     ^^^^^^^^^^^^^ 

semantic error: Type-spec cannot contain an asterisk for a type parameter
  --> tests/errors/continue_compilation_3.f90:66:14
   |
66 |     print *, [character(*) :: "a", "b", "ball", "cat"]
   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 

semantic error: Type-spec cannot contain an asterisk for a type parameter
  --> tests/errors/continue_compilation_3.f90:67:14
   |
67 |     print *, [character(*) :: "a2", "b2", "ball2", "cat2"]
   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 

semantic error: Different shape for array assignment on dimension 1(3 and 2)
  --> tests/errors/continue_compilation_3.f90:69:5
   |
69 |     x1 = reshape([1,2,3,4],[2,2])
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 

semantic error: Different shape for array assignment on dimension 1(3 and 1)
  --> tests/errors/continue_compilation_3.f90:70:5
   |
70 |     x1 = reshape([1,2,3,4],[1,2])
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 

semantic error: Arithmetic if (x) requires an integer or real for `x`
  --> tests/errors/continue_compilation_3.f90:74:9
   |
74 |     if ("yy") 1, 2, 3
   |         ^^^^ 

semantic error: Incorrect number of arguments passed to the 'size' intrinsic. It accepts at least 1 and at most 3 arguments.
  --> tests/errors/continue_compilation_3.f90:81:9
   |
81 |     i = size(a1, 1, 4, kind=4)
   |         ^^^^^^^^^^^^^^^^^^^^^^ 

semantic error: Incorrect number of arguments passed to the 'size' intrinsic. It accepts at least 1 and at most 3 arguments.
  --> tests/errors/continue_compilation_3.f90:82:9
   |
82 |     i = size()
   |         ^^^^^^ 

semantic error: Cannot assign to a constant variable
  --> tests/errors/continue_compilation_3.f90:84:5
   |
84 |     x3 = 1
   |     ^^^^^^ assignment here
   |
17 |     integer, parameter :: x3 = 2
   |                           ~~~~~~ declared as constant

semantic error: The end variable of the data implied do loop must be constants
  --> tests/errors/continue_compilation_3.f90:87:22
   |
87 |     data(a1(i), i=1, k) / 1, 2, 3 /
   |                      ^ 

semantic error: The increment variable of the data implied do loop must be a constant
  --> tests/errors/continue_compilation_3.f90:89:25
   |
89 |     data(a1(i), i=1, 3, k) / 1, 2, 3 /
   |                         ^ 

semantic error: The start variable of the data implied do loop must be constants
  --> tests/errors/continue_compilation_3.f90:91:19
   |
91 |     data(a1(i), i=k, 3) / 1, 2, 3 /
   |                   ^ 

semantic error: Variable 'foo' is not declared
  --> tests/errors/continue_compilation_3.f90:94:9
   |
94 |     i = foo
   |         ^^^ 'foo' is undeclared

semantic error: Invalid argument `end` supplied
  --> tests/errors/continue_compilation_3.f90:96:5
   |
96 |     rewind(end="world")
   |     ^^^^^^^^^^^^^^^^^^^ 

semantic error: same_type_as is not implemented yet
  --> tests/errors/continue_compilation_3.f90:98:14
   |
98 |     print *, same_type_as(1.0, 1.0)
   |              ^^^^^^^^^^^^^^^^^^^^^^ 

semantic error: Kind of all the arguments of Sign must be the same
   --> tests/errors/continue_compilation_3.f90:100:14
    |
100 |     print *, sign(1, 1_8)
    |              ^^^^^^^^^^^^ 

semantic error: Argument 1 of dabs must be of double precision real type
   --> tests/errors/continue_compilation_3.f90:102:13
    |
102 |     print*, dabs(1)
    |             ^^^^^^^ 

semantic error: Argument of `sqrt` has a negative argument
   --> tests/errors/continue_compilation_3.f90:104:14
    |
104 |     print *, sqrt(-1.0)
    |              ^^^^^^^^^^ 

semantic error: Binary numeric operators cannot be used on strings
   --> tests/errors/continue_compilation_3.f90:106:14
    |
106 |     print *, "a" + "b"
    |              ^^^^^^^^^ help: use '//' for string concatenation

semantic error: The first index in string section is less than 1
   --> tests/errors/continue_compilation_3.f90:109:19
    |
109 |     print*, "s:", s(-1:4)
    |                   ^^^^^^^ 

semantic error: Substring `start` is less than one
   --> tests/errors/continue_compilation_3.f90:111:13
    |
111 |     print*, s1(-2:6)
    |             ^^^^^^^^ 

semantic error: Substring end index exceeds the string length
   --> tests/errors/continue_compilation_3.f90:114:13
    |
114 |     print*, s1(1: 9)
    |             ^^^^^^^^ 

semantic error: Substring end index at must be of type integer
   --> tests/errors/continue_compilation_3.f90:116:18
    |
116 |     print*, s1(1:5.2)
    |                  ^^^ 

semantic error: Substring start index at must be of type integer
   --> tests/errors/continue_compilation_3.f90:118:16
    |
118 |     print*, s1(1.1:5)
    |                ^^^ 

semantic error: Substring stride must be of type integer
   --> tests/errors/continue_compilation_3.f90:120:19
    |
120 |     print*, s(1:5:2.2)
    |                   ^^^ 

semantic error: Type mismatch in assignment, the types must be compatible
   --> tests/errors/continue_compilation_3.f90:122:5
    |
122 |     x = "x"
    |     ^   ^^^ type mismatch (integer and string)

semantic error: Type mismatch in binary operator, the types must be compatible
   --> tests/errors/continue_compilation_3.f90:124:9
    |
124 |     x = 5 + "x"
    |         ^   ^^^ type mismatch (integer and string)

semantic error: Type mismatch in assignment, the types must be compatible
   --> tests/errors/continue_compilation_3.f90:124:5
    |
124 |     x = 5 + "x"
    |     ^       ^^^ type mismatch (integer and string)

semantic error: Subroutine `bpe` called as a function
   --> tests/errors/continue_compilation_3.f90:127:9
    |
127 |     i = bpe()
    |         ^^^^^ 

semantic error: Variable 'xx' is not declared
   --> tests/errors/continue_compilation_3.f90:128:14
    |
128 |     print *, xx
    |              ^^ 'xx' is undeclared

semantic error: Variable 'test_re' is not declared
   --> tests/errors/continue_compilation_3.f90:129:5
    |
129 |     test_re = 1245.13
    |     ^^^^^^^ 'test_re' is undeclared

semantic error: Array reference is not allowed on scalar variable
   --> tests/errors/continue_compilation_3.f90:130:5
    |
130 |     c(1) = 1
    |     ^^^^ 

semantic error: Argument of `size` must be an array
   --> tests/errors/continue_compilation_3.f90:140:18
    |
140 |         print *, size(bpe)
    |                  ^^^^^^^^^ 

semantic error: Variable 'd' is not declared
   --> tests/errors/continue_compilation_3.f90:141:15
    |
141 |         bpe = d
    |               ^ 'd' is undeclared

semantic error: Assignment to subroutine is not allowed
   --> tests/errors/continue_compilation_3.f90:141:9
    |
141 |         bpe = d
    |         ^^^ 
